### DISCUSSION TITLE
Workflow Orchestration Stuck in Running state for hours

### DISCUSSION THREAD

**Discussion by gorillapower** [Azure/azure-functions-durable-extension#2062] [Q&A] [answered]
One of our Workflows occasionally never complete. We are seeing this once or twice a day currently (2 out of every +-50 runs)

The current UTC time is 10:32 so the function has been 'stuck' in a running state since 02:56 (+7 hours)
Can anyone guide me as to how to debug this issue? I dont see any messages in the control or worker queues.

![image](https://user-images.githubusercontent.com/8879065/151952908-6311d932-f216-454b-b79e-b269bd0ea718.png)

InstanceId: 845a0f2b-d3e0-4d26-9237-2c7fb3c73271 (West Europe) *Please note there will be two functions with the same instanceId in our environment, this is by design.
Plan: Consumption Plan
Runtime: nodejs
FUNCTIONS_EXTENSION_VERSION: ~3
FUNCTIONS_WORKER_RUNTIME_VERSION : 12

package.json
```{
  "dependencies": {
    "azure-storage": "^2.10.1",
    "durable-functions": "1.4.3",
    "moment": "^2.22.2",
    "uuid": "^3.3.2",
    "vm2": "3.6.3"
  }
} 
```

**Comment by cgillum** [comment, marked as answer]
For these types of issues, I recommend the following strategy:
- Use the [Diagnose and Solve Problems tool](https://docs.microsoft.com/en-us/azure/azure-functions/functions-diagnostics) in the Azure Portal and search for "Durable Functions" to run some automated analysis on your app. Certain types of issues will be identified by this tool and can provide guidance on how to investigate further. One of the most common resolutions is upgrading to the latest version of Durable Functions, since we regularly release improvements.
- Enable [Durable Task Framework logging](https://docs.microsoft.com/en-us/azure/azure-functions/durable/durable-functions-diagnostics?tabs=javascript#durable-task-framework-logging) and look for warnings or errors that occurred around the time at which the problem started occurring. In almost all cases there is some warning or error that will provide you with hints about what's causing the issue.
- If the above two don't help, either open an issue or file a support request with Azure Functions. The latter will ensure that you get some sort of resolution within a particular SLA.

In your case, looking just at the dependencies, I suggest upgrading to the latest version of `durable-functions`, which is **1.5.4**. @davidmrdavid has made a lot of improvements to this library in recent months. You'll also want to make sure that you're on the latest extension bundles versions.
